### PR TITLE
Add support for kafka 2.1.0

### PR DESCRIPTION
**Pull request by renatomefi**

### CODE REVIEW

File: src/main/java/com/salesforce/mirus/Mirus.java [hunk 1]
```diff
@@ -135,10 +136,13 @@ private static void startConnect(Map<String, String> workerProps) {
     StatusBackingStore statusBackingStore =
         new KafkaStatusBackingStore(time, worker.getInternalValueConverter());
     statusBackingStore.configure(configWithClientIdSuffix(workerProps, "status"));
+    WorkerConfigTransformer configTransformer = worker.configTransformer();
```
Comment (pdavidson100) [inline src/main/java/com/salesforce/mirus/Mirus.java]: No need to declare `configTransformer` here, just use `worker.configTransformer()` below.

File: pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <properties>
         <dist.dir>${project.build.directory}/dist</dist.dir>
         <java.numeric.version>1.8</java.numeric.version>
-        <kafka.version>1.1.0</kafka.version>
+        <kafka.version>2.1.0</kafka.version>
```
Comment (pdavidson100) [inline pom.xml]: This change causes a few new shader warnings.  I suggest you update the `maven-shade-plugin` to version `3.2.1 `, and also add this exclusion to the `connect-runtime` dependency:
```    
           <exclusions>
                <exclusion>
                    <groupId>org.glassfish.hk2.external</groupId>
                    <artifactId>javax.inject</artifactId>
                </exclusion>
            </exclusions>
```

This still leaves one warning about breaking module encapsulation, which seems to be unavoidable when using module dependency jars with the shader.  Long term we may need to consider a different way of packaging dependencies, but this seems to be OK for now.

**Comment by renatomefi** [comment]
Hey @pdavidson100 
I've addressed the issues! Where are you seeing those warnings tho? I'm a java newbie so this escaped my eyes!

I hope the PR is fine now!

**Comment by pdavidson100** [comment]
> Hey @pdavidson100
> I've addressed the issues! Where are you seeing those warnings tho? I'm a java newbie so this escaped my eyes!

The warnings are visible in the Maven logs when you run `mvn package -P all`, which generates a "fat" jar.

**Pull request merged**